src/DataCollection.Shared/ViewModels/AttachmentsViewModel.cs

```diff
@@ -0,0 +1,200 @@
+using Esri.ArcGISRuntime.ExampleApps.DataCollection.Shared.Models;
+using Esri.ArcGISRuntime.ExampleApps.DataCollection.Shared.Properties;
+using Esri.ArcGISRuntime.Mapping.Popups;
+using System;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Input;
+
+namespace Esri.ArcGISRuntime.ExampleApps.DataCollection.Shared.ViewModels
+{
+    public class AttachmentsViewModel : BaseViewModel
+    {
+        private PopupManager _popupManager;
+
+        public PopupManager PopupManager
+        {
+            get => _popupManager;
+            set
+            {
+                _popupManager = value;
+                OnPropertyChanged();
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AttachmentsViewModel"/> class.
+        /// </summary>
+        public AttachmentsViewModel(PopupManager popupManager, FeatureTable featureTable)
+        {
+            if (popupManager != null)
+            {
+                AttachmentManager = popupManager.AttachmentManager;
+
+                _popupManager = popupManager;
+
+                Attachments = new ObservableCollection<AttachmentWithThumbnail>();
+
+                LoadAttachments();
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the collection of attachments to be displayed 
+        /// </summary>
+        public ObservableCollection<AttachmentWithThumbnail> Attachments { get; private set; }
+
+        /// <summary>
+        /// Gets or sets the attachment manager for the feature
+        /// </summary>
+        public PopupAttachmentManager AttachmentManager { get; private set; }
+
+        private ICommand _openAttachmentCommand;
+
+        /// <summary>
+        /// Gets the command to open the attachment the user tapped on
+        /// </summary>
+        public ICommand OpenAttachmentCommand
+        {
+            get
+            {
+                return _openAttachmentCommand ?? (_openAttachmentCommand = new DelegateCommand(
+                    async (x) =>
+                    {
+                        if (x != null && x is PopupAttachment attachment)
+                        {
+                            if (attachment.LoadStatus != LoadStatus.Loaded)
+                            {
+                                await attachment.LoadAsync();
+                            }
+
+                            // create file name from runtime attachment filename hash and the actual attachment name
+                            var runtimeFileHash = attachment.Filename.Split('.').FirstOrDefault();
+                            var newFileName = runtimeFileHash + "." + attachment.Name;
+
+                            // if the file exists, rename it to contain the proper extension
+                            if (File.Exists(attachment.Filename))
```

Comment:
I realize the comment is misleading. The file already exists on disk because it's downloaded when the attachment is loaded. I'm just checking too see if it exists.

Reply:
The reason my comment mentioned the `proper extension` was because the file that gets downloaded has a random 3 letter extension rather than the actual file extension. For example, `tree.jpg` ens up on disk as `i5ilqfop.i1g`